pkg/gardenlet/controller/shoot/shoot_control.go

```diff
@@ -557,6 +520,141 @@ func (c *Controller) durationUntilNextShootSync(shoot *gardencorev1beta1.Shoot)
 	return syncPeriod
 }
 
+func (c *Controller) updateShootStatusAndRequeueOnSyncError(shoot *gardencorev1beta1.Shoot, err error) (reconcile.Result, error) {
+	message := fmt.Sprintf("Shoot cannot be synced with Seed: %v", err)
+	c.recorder.Event(shoot, corev1.EventTypeNormal, gardencorev1beta1.EventOperationPending, message)
+
+	_, updateErr := kutil.TryUpdateShootStatus(c.k8sGardenClient.GardenCore(), retry.DefaultRetry, shoot.ObjectMeta,
+		func(shoot *gardencorev1beta1.Shoot) (*gardencorev1beta1.Shoot, error) {
+			shoot.Status.LastOperation.Description = message
+			shoot.Status.LastOperation.LastUpdateTime = metav1.NewTime(time.Now().UTC())
+			return shoot, nil
+		},
+	)
+	if updateErr != nil {
+		return reconcile.Result{}, utilerrors.WithSuppressed(err, updateErr)
+	}
+
+	return reconcile.Result{
+		Requeue:      true,
+		RequeueAfter: 15 * time.Second, // prevent ddos-ing the seed
+	}, nil
+}
+
+func (c *Controller) updateShootStatusOperationStart(shoot *gardencorev1beta1.Shoot, shootNamespace string) (*gardencorev1beta1.Shoot, error) {
```

Comment:
FYI @swilen-iwanow, you might want to use these functions as part of #2012 as well (instead of duplicating (parts of) them again)